modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -326,6 +326,63 @@ func (suite *KeeperTestSuite) TestChanUpgradeTry() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestWriteUpgradeTry() {
```

Comment:
not sure if this is the type of unit testing you were thinking of @colin-axner

Reply:
yup, lgtm!